strongbox/strongbox
# Issue-840: Create an RPM distribution

Fixes #840. Done by @v3rtumnus!

Rebased @v3rtumnus 's #855.

@v3rtumnus ,

Sorry for getting back to this with such a big delay (it's not typical for our project), but we had more pressing problems with our Spring Boot migration, as well as some issues with the parallelization of our tests. I have rebased your original pull request and pushed it to a different branch.

@steve-todorov , @v3rtumnus : Would you like to try this out?

@steve-todorov : We will have to come up with a build process in Jenkins that either runs separately on a CentOS/Opensuse node after `strongbox` has been built, or make the Jenkins nodes have `rpmbuild` on them.

__Tasks:__
* [x] Rebase #855 onto the `master`.
* [x] Apply any required fixes to the rebased code, caused by changes from the `master`.
* [x] Update the version of the `maven-assembly-plugin` in the `strongbox-parent` to `3.1.1`.
* [x] Make the `rpm-maven-plugin` output the `rpm` under `target`, (instead of `target/rpm/strongbox-distribution/RPMS/noarch`).
* [x] Attach the produced `rpm` as a Maven artifact.
* [x] Fix the Maven plugin version warnings.
* [x] ~Drop the timestamp from the RPM's version.~ [Not possible].
* [x] Re-structure the `rpm` to use `/opt`.
* [x] The `lib` directory contains a ton of dependencies which shouldn't be there.
* [x] Add a the `rpm` build in a Maven profile, which is triggered by invoking `mvn clean install -Pbuild-rpm`.
* [x] ~Make the `finalName` of the RPM follow the RPM naming when deploying to Github.~ [Not possible]
* [x] Sort out variable comments raised by @steve-todorov.

__Useful `rpm` commands:__
* List installed packages:
```
rpm -qa 
```
* Check if the `strongbox` package is installed:
```
rpm -qa | grep strongbox
```
* Install package:
```
rpm -ivh strongbox-distribution-1.0-SNAPSHOT.rpm
```
* List package contents
```
rpm -qlp strongbox-distribution-1.0-SNAPSHOT20190309035450.x86_64
```
* Remove package:
```
rpm -e strongbox-distribution-1.0-SNAPSHOT20190309035450.x86_64
```
